src/main/java/teammates/storage/sqlapi/FeedbackSessionsDb.java

```diff
@@ -235,4 +237,27 @@ public List<FeedbackSession> getFeedbackSessionEntitiesForCourseStartingAfter(St
 
         return HibernateUtil.createQuery(cr).getResultList();
     }
+
+    /**
+     * Gets a list of undeleted feedback sessions which start within the last 2 hours
+     * and possibly need an open email to be sent.
+     */
+    public List<FeedbackSession> getFeedbackSessionsPossiblyNeedingOpenEmail() {
+        return getFeedbackSessionEntitiesPossiblyNeedingOpenEmail().stream()
+                .filter(session -> session.getDeletedAt() == null)
+                .collect(Collectors.toList());
+    }
+
+    private List<FeedbackSession> getFeedbackSessionEntitiesPossiblyNeedingOpenEmail() {
+        CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
+        CriteriaQuery<FeedbackSession> cr = cb.createQuery(FeedbackSession.class);
+        Root<FeedbackSession> root = cr.from(FeedbackSession.class);
+
+        cr.select(root)
+                .where(cb.and(
+                    cb.greaterThan(root.get("startTime"), TimeHelper.getInstantDaysOffsetFromNow(-2)),
+                    cb.equal(root.get("isOpenEmailSent"), false)));
```

Comment:
Yes I agree that we maybe should be checking for this as well, but the original implementation (using datastore) does not include this. I'm reluctant to make changes to the logic as it might cause regressions.

Reply:
This is reasonable. Maybe we can create a separate issue for this then.